sympy/stats/stochastic_process_types.py

```diff
@@ -171,7 +175,11 @@ def __getitem__(self, time):
         if time not in self.index_set:
             raise IndexError("%s is not in the index set of %s"%(time, self.symbol))
         idx_obj = Indexed(self.symbol, time)
-        pspace_obj = StochasticPSpace(self.symbol, self)
+        try:
+            distribution = self.distribution
+        except AttributeError:
+            distribution = None
```

Comment:
Can you do this without try-except block?

Reply:
I tried it, but then it would raise error for DTMC and CTMC,
Also using the other way around by adding the following would lead to raise test_args errors in MarkovProcess, DTMC, CTMC
```
# In Stochastic Process
def __new__(cls, sym, state_space=S.Reals, distribution=None, **kwargs):
        sym = _symbol_converter(sym)
        state_space = _set_converter(state_space)
        return Basic.__new__(cls, sym, state_space, distribution)

@property
def distribution(self):
       return self.args[2]
```

Reply:
Also adding distribution is neccesary at that line as otherwise, StochasticPSpace().distribtuion would get assigned None.

Reply:
What about the following? Do tests pass on this?
```python
def __getitem__(self, time):
        """
        For indexing discrete time stochastic processes.

        Returns
        =======

        RandomIndexedSymbol
        """
        if time not in self.index_set:
            raise IndexError("%s is not in the index set of %s"%(time, self.symbol))
        idx_obj = Indexed(self.symbol, time)
        distribution = None
        if hasattr(self, 'distribution'):
            distribution = self.distribution
        pspace_obj = StochasticPSpace(self.symbol, self, distribution)
        return RandomIndexedSymbol(idx_obj, pspace_obj)
```

Reply:
Yes, it worked.